src/main/java/seedu/address/logic/commands/issue/MarkIssueCommand.java

```diff
@@ -0,0 +1,66 @@
+package seedu.address.logic.commands.issue;
+
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.model.Model.PREDICATE_SHOW_ALL_ISSUES;
+
+import java.util.List;
+
+import seedu.address.commons.core.Messages;
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.commands.CommandResult;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.issue.Issue;
+import seedu.address.model.issue.Status;
+import seedu.address.ui.Ui;
+
+/**
+ * Command to mark an issue as complete or incomplete
+ */
+public class MarkIssueCommand extends IssueCommand {
+
+    public static final String COMMAND_FLAG = "-m";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + " " + COMMAND_FLAG
+            + ": Marks the issue identified by the issue id "
+            + "as complete if it is currently incomplete, or incomplete if it is currently completed"
+            + "Parameters: ISSUE_ID (must be a positive integer) "
+            + "Example: " + COMMAND_WORD + " "
+            + COMMAND_FLAG + " 1";
+
+
+    public static final String MESSAGE_SUCCESS = "Issue marked: %1$s";
+
+    private final Index targetIndex;
+
+    public MarkIssueCommand(Index targetIndex) {
+        this.targetIndex = targetIndex;
+    }
+
+    @Override
+    public CommandResult execute(Model model, Ui ui) throws CommandException {
+        requireNonNull(model);
+        List<Issue> lastShownList = model.getFilteredIssueList();
+
+        for (Issue i : lastShownList) {
```

Comment:
Logic could be moved to somewhere more appropriate - the following suggestions are possible:
1. Find the `Issue` by ID in `IssueCommandParser`, then either create a new `Issue` instance with all fields the same except `Status` there, or do a `setStatus()`. Pass the edited `Issue` to `MarkCommand` which calls `model.setIssue()`.
2. Move the logic down to `ModelManager` - although this still leads to Law of Demeter violations